### PR TITLE
gltf assign final material parity

### PR DESCRIPTION
**Pull request by HexaField**
## References
closes #_insert number here_

### CODE REVIEW

File: packages/engine/src/gltf/GLTFState.tsx [hunk 1]
```diff
@@ -1136,6 +1140,25 @@ const MaterialInstanceReactor = (props: {
     else materialInstance.uuid.set([materialUUID])
   }, [materialEntity, primitive.material])
 
+  const material = useOptionalComponent(materialEntity, MaterialStateComponent)?.material
+  const useDerivativeTangents = primitive.attributes.tangent === undefined
+  const useVertexColors = primitive.attributes.color !== undefined
+  const useFlatShading = primitive.attributes.normal === undefined
```
Comment (AidanCaruso) [inline packages/engine/src/gltf/GLTFState.tsx]: These will always evaluate to undefined because the primitive attribute properties are in all caps.
image

Comment (HexaField) [reply packages/engine/src/gltf/GLTFState.tsx]: the caps are once they are translated to threejs, in this form they are in the gltf format, which is lower case.

Comment (HexaField) [reply packages/engine/src/gltf/GLTFState.tsx]: hmm ok you may be right :/

Comment (HexaField) [reply packages/engine/src/gltf/GLTFState.tsx]: ahh, in the threejs loader, it was looking for parameters on the geometry. this is now on the gltf document. my mistake. good catch.

Comment (HexaField) [reply packages/engine/src/gltf/GLTFState.tsx]: Screenshot from 2024-11-15 09-55-33
and look at that, vertex colors working!

**Review by AidanCaruso — APPROVED** [review]

**Pull request merged**